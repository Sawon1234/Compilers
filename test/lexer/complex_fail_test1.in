# Mapper

interface Mapper<T,S>
{
  fun map(it : Iterable<T>) : List<S>
  {
    out := [];
    for(t in it)
    {
      out := [elemmap(t)] ++ out;
    }
    return out;
  }
  fun elemmap(elem : T) : S;
}