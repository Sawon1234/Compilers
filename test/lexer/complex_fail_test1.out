interface Name < Name , Name > { fun name ( name : Name < Name > ) : Name < Name > { name := [ ] ; for ( name in name ) { name := [ name ( name ) ] ++ name ; } return name ; } fun name ( name : Name ) : Name ; }